apollo-router/src/plugin/mod.rs

```diff
@@ -216,16 +225,30 @@ impl PluginInit<serde_json::Value> {
     where
         T: for<'de> Deserialize<'de>,
     {
-        PluginInit::try_builder()
-            .config(self.config)
-            .supergraph_schema(self.supergraph_schema)
-            .supergraph_schema_id(self.supergraph_schema_id)
-            .supergraph_sdl(self.supergraph_sdl)
-            .subgraph_schemas(self.subgraph_schemas)
-            .notify(self.notify.clone())
-            .license(self.license)
-            .and_full_config(self.full_config)
-            .build()
+        // Use conditional building to avoid passing None to previous_config
```

Comment:
(isn't there a `.and_previous_config(opt_config)` for that?)

Reply:
You are right [96f706a](https://github.com/apollographql/router/pull/8192/commits/96f706a9a6ce179de7e8e4d17c76e8e3739d136c)